src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,728 @@
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public TallBuildingCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.bufferDistanceMeters = this.configurationValue(configuration, "bufferDistanceMeters",
+                BUFFER_DISTANCE_DEFAULT);
+        this.minDatasetSizeForStatsComputation = this.configurationValue(configuration,
+                "minDatasetSizeForStatsComputation", MIN_DATASET_SIZE_DEFAULT);
+        this.maxLevelTagValue = this.configurationValue(configuration, "maxLevelTagValue",
+                MAX_LEVEL_TAG_VALUE_DEFAULT);
+        this.outlierMultiplier = this.configurationValue(configuration, "outlierMultiplier",
+                OUTLIER_MULTIPLIER_DEFAULT);
+        this.invalidHeightCharacters = new HashSet<>(this.configurationValue(configuration,
+                "invalidHeightCharacters", INVALID_CHARACTER_DEFAULT));
+        this.magicNumber3 = this.configurationValue(configuration, "magicNumbers.3", THREE_DEFAULT);
+        this.magicNumber4 = this.configurationValue(configuration, "magicNumbers.4", FOUR_DEFAULT);
+        this.magicNumberOneQuarter = this.configurationValue(configuration,
+                "magicNumbers.oneQuarter", ONE_QUARTER_DEFAULT);
+        this.magicNumberThreeQuarters = this.configurationValue(configuration,
+                "magicNumbers.threeQuarters", THREE_QUARTER_DEFAULT);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        final Map<String, String> tags = object.getTags();
+        return !this.isFlagged(object.getIdentifier())
+                && ((object instanceof Area)
+                        || (object instanceof Relation && ((Relation) object).isMultiPolygon()))
+                && (this.isBuildingOrPart(object) || this.isBuildingRelationMember(object))
+                && (this.hasHeightTag(tags) || this.hasBuildingLevelTag(tags));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        markAsFlagged(object.getOsmIdentifier());
```

Comment:
Looks like the OsmIdentifier is being marked as flagged and the atlasIdentifier is checked if flagged in validCheckForObject.

Reply:
addressed